Caskets: Split area loot

Omnione opened the pull request:
This allows areas with split mob levels to assign loot to caskets,
previously left out from a mistake on a rebase.


<!-- place 'x' mark between square [] brackets to affirm: -->
**_I affirm:_**
- [X] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [X] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

Omnione committed:
Caskets: allow split zone loot

Rebase took some code out for split mob levels,
This PR will allow loot to once again be assigned from zones with mob levels that greatly differ.

Omnione committed:
Left an old unneeded bool in

Removed.

ibm2431 commented on scripts/globals/caskets.lua:
```diff
@@ -401,10 +402,23 @@ function getDrops(npc, dropType, zoneId)
         local tempCount    = 1
         local randomTable  = {1,3,1,2,1,2,1,1,3,1,2,1}
 
+        if zoneId == 172 or zoneId == 173 or zoneId == 190 or
+           zoneId == 191 or zoneId == 193 or zoneId == 194 or
+           zoneId == 196 or zoneId == 198 then
```
Only thing I think might be improved on is defining these zones as a Set in the caskets global:
```lua
casketInfo.splitZones = set{
    tpz.zone.ZERUHN_MINES,
    tpz.zone.KORROLOKA_TUNNEL
}
```

And then your check here becomes:
`if casketInfo.splitZones[zoneId] then`

Omnione replied:
yep sounds good

ibm2431 commented on scripts/globals/caskets.lua:
```diff
@@ -436,10 +450,22 @@ function getDrops(npc, dropType, zoneId)
         local itemCount    = 1
         local randomTable  = {1,4,1,3,1,1,2,1,3,1,2,1}
 
+        if zoneId == 172 or zoneId == 173 or zoneId == 190 or
+           zoneId == 191 or zoneId == 193 or zoneId == 194 or
+           zoneId == 196 or zoneId == 198 then
```
And with a Set already defined, the same check can be used here too: `if casketInfo.splitZones[zoneId] then`

Omnione replied:
sounds legit, ill get on that

Omnione committed:
Caskets: adjustment requests

• added a new table for split zones, and can now use casketInfo.splitZones[zoneId]

Omnione commented:
Ok tested the changes requested, all seems good, didnt even know we had set{}, i like that, ill have to remember that one.

ibm2431 commented:
I slipped it in a while back. I've kept an eye out for opportunities to share its joys ever since!

ibm2431 reviewed (approved):
Thanks, Omnione!

Omnione commented:
lol nice
